src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -16,10 +16,31 @@
              [interface :as i]
              [util :as sync-util]]
             [metabase.util :as u]
-            [metabase.util.schema :as su]
+            [metabase.util
+             [schema :as su]
+             [metrics :as um]]
+            [metrics
+             [histograms :refer [defhistogram update!]]
+             [meters :refer [defmeter mark!]]
+             [timers :refer [deftimer time!]]]
             [schema.core :as s]
             [toucan.db :as db]))
 
+(deftimer um/reg sync-fields-timer)
```

Comment:
I used timers, meters and histograms here. They can be def'd like I did here or identified by a string in the central registry.

Timers will give you counts and average timings for each invocation. Meters just count things as they go by and histograms will mark an event with a value.